_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: post
+name: docker-meets-continuous-deployment
+title: Docker meets Continuous Deployment
+date: 2016-03-01 00:00:00 +00:00
+author: Robert Firek
+image:
+    src: /assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment.png
+    attribution:
+        text: Track icon by Freepik (CC 3.0 BY)
+        href: http://www.freepik.com
+tags:
+- Continuous Deployment
+- Docker
+- TeamCity
+--- 
+
+About one year ago I had my first contact with Docker. This new kid on the block promised to relieve our poor computers from installation of all tools, languages, dependencies and operating systems. Isolated run environments emerged on developers computers.
+
+A conservative approach of my ops teammates didn't change my joy of using Docker. Despite many people describing Docker as tool written by developers for development our industry found new ways of using images and containers. Images of our applications and services became units of deployment for tools like Kubernetes, Docker Swarm or Marathon.
+
+But how these images are created?
+
+## Setting The Scene
+
+From a developer's perspective any application is manifested by its code, but there is still a long way to go before this code finds its way to a production environment. I want to show you how you this process can be easier with **Docker** and a **Continuous Deployment** pipeline. 
+
+First we need a small application with a HTTP API that we can call after it is deployed. Let's assume that are using **Gradle** to build the application and **TeamCity** as a Continuous Integration server.
+
+We need to have **Docker** installed on each **TeamCity** build agent. We will also use this machine to run our application. In a real project we wouldn't install **TeamCity** agents on all machines. Instead we should use tools like Kubernetes that will take care of the application distribution.
+ 
+## Build
+
+We need to build something before we will be able to create any docker image. The first step in our **Continuous Deployment** pipeline would be to build our application. In this step we will download the source code, run all **tests** and produce an **artifact** containing all elements required to start and run our application.
+
```

Comment:
We need to build something before we will be able to create any docker image. The first step in our **Continuous Deployment** pipeline would be to build our application. -> As a first step in our **Continuous Deployment** pipeline, before we even think about Docker images, we need to build our application.